### PR TITLE
OCPBUGS-38002: Updated EggressIPconfig object to match API description

### PR DESCRIPTION
**Pull request by dfitzmau**
Version(s):
4.12+

Issue:
[OCPBUGS-38002](https://issues.redhat.com/browse/OCPBUGS-38002)

Link to docs preview:
[EgressIPconfig object](https://80138--ocpdocs-pr.netlify.app/openshift-enterprise/latest/networking/ovn_kubernetes_network_provider/configuring-egress-ips-ovn.html#nw-egress-ips-config-object_configuring-egress-ips-ovn)

- [x] SME has approved this change (Martin Kennelly).
- [x] QE has approved this change (Zhanqi, huiran0826, or jechen0648 ).

Additional information:

### CODE REVIEW

File: modules/nw-egress-ips-config-object.adoc [hunk 1]
```diff
@@ -28,8 +31,5 @@ spec:
         routingViaHost: false
       genevePort: 6081
 ----
-<1> The `egressIPConfig` holds the configurations for the options of the `EgressIP` object. Changing these configurations allows you to extend the `EgressIP` object.
-
-<2> The value for `reachabilityTotalTimeoutSeconds` accepts integer values from `0` to `60`. A value of 0 disables the reachability check of the egressIP node. Values of `1` to `60` correspond to the duration in seconds between probes sending the reachability check for the node.
-
-
+<1> The `egressIPConfig` holds the configurations for the options of the `EgressIP` object. By changing these configurations, you can extend the `EgressIP` object (in what way?).
```
Comment (dfitzmau) [inline modules/nw-egress-ips-config-object.adoc:34]: you can extend the `EgressIP` object to where?

File: modules/nw-egress-ips-config-object.adoc [hunk 2]
```diff
@@ -28,8 +31,5 @@ spec:
...
+<2> The value for `reachabilityTotalTimeoutSeconds` accepts integer values from `0` to `60`. A value of `0` disables the reachability check of the egressIP node. Values of `1` to `60` correspond to the duration in seconds between probes sending the reachability check for the node.
```
Comment (dfitzmau) [inline modules/nw-egress-ips-config-object.adoc:35]: Does:

> Values of `1` to `60` correspond to the duration in seconds between probes sending the reachability check for the node.

Those this sentence indicate the "interval" situation as mentioned in the Jira ticket? https://issues.redhat.com/browse/OCPBUGS-38002

File: modules/nw-egress-ips-config-object.adoc [hunk 3]
```diff
@@ -28,8 +31,5 @@ spec:
         routingViaHost: false
       genevePort: 6081
 ----
-<1> The `egressIPConfig` holds the configurations for the options of the `EgressIP` object. Changing these configurations allows you to extend the `EgressIP` object.
-
-<2> The value for `reachabilityTotalTimeoutSeconds` accepts integer values from `0` to `60`. A value of 0 disables the reachability check of the egressIP node. Values of `1` to `60` correspond to the duration in seconds between probes sending the reachability check for the node.
-
-
+<1> The `egressIPConfig` holds the configurations for the options of the `EgressIP` object. By changing these configurations, you can extend the `EgressIP` object.
+<2> The value for `reachabilityTotalTimeoutSeconds` accepts integer values from `0` to `60`. A value of `0` disables the reachability check of the egressIP node. Setting a value from `1` to `60` corresponds to the duration in seconds for a probe to send the reachability check for the node.
```
Comment (dfitzmau) [inline modules/nw-egress-ips-config-object.adoc:35]: "for the node" (on behalf) or "to the node"?

**Review by martinkennelly — APPROVED** [review]
lgtm (from networking team)

**Comment by anuragthehatter** [comment]
cc @zhaozhanqi

**Comment by zhaozhanqi** [comment]
@huiran0826 @jechen0648 Could you help take a look this?

File: modules/nw-egress-ips-config-object.adoc [hunk 4]
```diff
@@ -4,11 +4,14 @@
...
 = EgressIPconfig object
-As a feature of egress IP, the `reachabilityTotalTimeoutSeconds` parameter configures the total timeout for checks that are sent by probes to egress IP nodes. The `egressIPConfig` object allows users to set the `reachabilityTotalTimeoutSeconds` `spec`. If the EgressIP node cannot be reached within this timeout, the node is declared down.
 
-You can increase this value if your network is not stable enough to handle the current default value of 1 second.
+As a feature of egress IP, the `reachabilityTotalTimeoutSeconds` parameter configures the EgressIP node reachability check total timeout in seconds. If the EgressIP node cannot be reached within this timeout, the node is declared down.
 
-The following YAML describes changing the `reachabilityTotalTimeoutSeconds` from the default 1 second probes to 5 second probes:
+You can set a value for the `reachabilityTotalTimeoutSeconds` in the configuration file for the `EgressIPconfig` object. Setting a large value might cause the EgressIP implementation to react slowly to node changes. The implementation reacts slowly for EgressIP nodes that have an issue and are unreachable.
```
Comment (JoeAldinger) [inline modules/nw-egress-ips-config-object.adoc:10]:
```suggestion
You can set a value for the `reachabilityTotalTimeoutSeconds` in the configuration file for the `egressIPConfig` object. Setting a large value might cause the EgressIP implementation to react slowly to node changes. The implementation reacts slowly for EgressIP nodes that have an issue and are unreachable. 
```

Comment (JoeAldinger) [reply modules/nw-egress-ips-config-object.adoc:10]: Um, a bit embarrassing but do mind grabbing my typos in this doc while you are here. I introduced them starting at the title. If not, no worries and I'll clean this up in a follow up PR. I think we should get these two, please. I _think_ I introduced this typo stemming from the title.

File: modules/nw-egress-ips-config-object.adoc [hunk 5]
```diff
@@ -4,11 +4,14 @@
...
 
-You can increase this value if your network is not stable enough to handle the current default value of 1 second.
+As a feature of egress IP, the `reachabilityTotalTimeoutSeconds` parameter configures the EgressIP node reachability check total timeout in seconds. If the EgressIP node cannot be reached within this timeout, the node is declared down.
 
-The following YAML describes changing the `reachabilityTotalTimeoutSeconds` from the default 1 second probes to 5 second probes:
+You can set a value for the `reachabilityTotalTimeoutSeconds` in the configuration file for the `EgressIPconfig` object. Setting a large value might cause the EgressIP implementation to react slowly to node changes. The implementation reacts slowly for EgressIP nodes that have an issue and are unreachable. 
+
+If you omit the `reachabilityTotalTimeoutSeconds` parameter from the `EgressIPconfig` object, the platform chooses a reasonable default value, which is subject to change over time. The current default is `1` second. A value of `0` disables the reachability check for the EgressIP node.
```
Comment (JoeAldinger) [inline modules/nw-egress-ips-config-object.adoc:12]:
```suggestion
If you omit the `reachabilityTotalTimeoutSeconds` parameter from the `egressIPConfig` object, the platform chooses a reasonable default value, which is subject to change over time. The current default is `1` second. A value of `0` disables the reachability check for the EgressIP node.
```

File: modules/nw-egress-ips-config-object.adoc (hunk 4 above)
Comment (JoeAldinger) [reply modules/nw-egress-ips-config-object.adoc:10]: I see it this way here: https://docs.openshift.com/container-platform/4.14/rest_api/operator_apis/network-operator-openshift-io-v1.html#spec-defaultnetwork-ovnkubernetesconfig-egressipconfig:~:text=Description-,egressIPConfig,-holds%20the%20configuration

**Review by JoeAldinger — COMMENTED** [review]
I think there are still two typos.

File: modules/nw-egress-ips-config-object.adoc (hunk 4 above)
Comment (dfitzmau) [reply modules/nw-egress-ips-config-object.adoc]: Not at all. Easily happens. My brain did not see these differences either until you pointed them out. Updated.

File: modules/nw-egress-ips-config-object.adoc (hunk 5 above)
Comment (dfitzmau) [reply modules/nw-egress-ips-config-object.adoc]: :heavy_check_mark:

**Pull request merged**